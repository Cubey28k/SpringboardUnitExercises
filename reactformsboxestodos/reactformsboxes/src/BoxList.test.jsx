import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import { it, expect } from "vitest";
import BoxList from "./BoxList";

function addBox(boxList, height = "2", width = "2", color = "peachpuff") {
    const heightInput = boxList.getByLabelText("Height");
    const widthInput = boxList.getByLabelText("Width");
    const backgroundInput = boxList.getByLabelText("Background Color");
    fireEvent.change(backgroundInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Click to add a new box.");
    fireEvent.click(button);
  }
  
  it("renders without crashing", function() {
    render(<BoxList />);
  });
  
  it("matches snapshot", function() {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("can add a new box", function() {
    const boxList = render(<BoxList />);
  
    // no boxes yet
    expect(boxList.queryByText("Remove this box")).not.toBeInTheDocument();
  
    addBox(boxList);
  
    // expect to see a box
    const removeButton = boxList.getByText("Remove this box");
    expect(removeButton).toBeInTheDocument();
    expect(removeButton.previousSibling).toHaveStyle(`
    width: 2em;
    height: 2em;
    background-color: rgb(255, 218, 185);
  `);
  
    // expect form to be empty
    expect(boxList.getAllByDisplayValue("")).toHaveLength(3);
  
    // expect(asFragment()).toMatchSnapshot();
  });
  
  it("can remove a box", function() {
    const boxList = render(<BoxList />);
    addBox(boxList);
  
    const removeButton = boxList.getByText("Remove this box");
  
    // click the remove button and the box should be gone
    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
  });
  