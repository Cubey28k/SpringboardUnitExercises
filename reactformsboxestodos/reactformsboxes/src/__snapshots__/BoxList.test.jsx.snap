// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="height"
          >
            Height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="width"
          >
            Width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="backgroundColor"
          >
            Background Color
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
        </div>
        <button
          id="newBoxButton"
        >
          Click to add a new box.
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
